charts/cluster-api-runtime-extensions-nutanix/addons/cni/cilium/values-template.yaml

```diff
@@ -36,8 +36,14 @@ envoy:
 {{- with .ControlPlane }}
 {{- range $key, $val := .metadata.annotations }}
 {{- if eq $key "controlplane.cluster.x-k8s.io/skip-kube-proxy" }}
+# Although its safe to set k8sServiceHost in all cases,
+# we're keeping it here to force a Cilium Pod rollout when enabling the kubeProxyReplacement feature.
```

Comment:
Thanks for the explanation

Reply:
I really don't like this. Are there any other ways to do this?

Reply:
How about we gate all of this https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/main/pkg/handlers/generic/lifecycle/cni/cilium/handler.go#L268-L282 based on if kube-proxy is installed and `capiutils.ShouldSkipKubeProxy(cluster)==true` and then trigger a DS rollout and wait for it to be restarted?

I actually noticed that none of this config, besides `k8sServiceHost` triggers new DS Pods to be recreated.